ocpi-toolkit-2.2.1/src/main/kotlin/com/izivia/ocpi/toolkit/modules/versions/services/VersionsService.kt

```diff
@@ -1,15 +1,36 @@
 package com.izivia.ocpi.toolkit.modules.versions.services
 
 import com.izivia.ocpi.toolkit.common.OcpiResponseBody
+import com.izivia.ocpi.toolkit.common.OcpiServerUnsupportedVersionException
 import com.izivia.ocpi.toolkit.modules.versions.VersionsInterface
 import com.izivia.ocpi.toolkit.modules.versions.domain.Version
+import com.izivia.ocpi.toolkit.modules.versions.domain.VersionDetails
+import com.izivia.ocpi.toolkit.modules.versions.domain.parseVersionNumber
 import com.izivia.ocpi.toolkit.modules.versions.repositories.VersionsRepository
 
 class VersionsService(
-    private val repository: VersionsRepository
+    private val repository: VersionsRepository,
+    private val baseUrl: String,
```

Comment:
We should take advantage of coroutines here and use a `baseUrlProvider` (that is what is done in `CredentialsServerService`)

```suggestion
    private val baseUrlProvider: suspend () -> String,
```

It allows anyone to use coroutine context to hold information about `baseUrl`. That is what we do in our multi tenant platform

Reply:
PR for #62 will contain examples, but we have something that looks like this:

```kotlin
VersionsService(
     repository = MyVersionsRepository(),
     baseUrlProvider = { config.ofCurrentTenant().baseUrl },
     // ...
)
```